planet-kaggle-blog.md

```diff
@@ -0,0 +1,69 @@
+
+### Discussion
+
+Many techniques that we implemented in Raster Vision and trialed did not increase our F2 score on the Amazon dataset. In several cases, we believe that the F2 score metric had unintentional effects on what constituted an accurate result. For example, we tried creating separate networks with different output layers for atmospheric tags versus the other tags. As there can be only one correct atmospheric tag, it seemed that a sigmoid layer would output better predictions than a softmax layer. Although counterintuitive to our expectation, this did not yield improved results. The metric places greater value on predicions with both correct and incorrect labels more than predictions which do not have the correct label at all. We also had experiments that artificially boosted the presence of rarer labels during training. We suspect that the F2 score simply did not place as much value on the performance for infrequent tags. Even if our predictions were more accurate for rare tags, it came at the cost of our overall F2 score across all the tags.
+
+Sometimes it takes a clever idea or an unorthodox breakthrough to push the accuracy of a predictive model past all its competitors. In this particular Amazon Kaggle competition, vanilla machine learning solutions were the best performing.
+
+Kaggle competitions are often won by very slim margins. In the case of the Amazon basin data, we spent a large portion of our development time trialing different methods of preprocessing and postprocessing the images. Kaggle competitions incentize improving your position on the public leaderboard. The score for the public leaderboard was calculated using 66% of the testing chips. The winner, however, is determined by the final private leaderboard score. This score uses the remaining 34% of the testing data and requires that the model appropriately combat overfitting to the public testing data too much. Since the private leaderboard performance is a blackbox until the competition deadline, generalizing the model is crucial to a high Kaggle rank. When the model approaches an accuracy that is hampered by the noise of the dataset itself, progression towards an incrementally better score can quickly become time, resource and computationally expensive. Many image processing, architectures or optmizers that did not improve our performance on the Amazon dataset remain useful for the scene tagging task and within Raster Vision.
```

Comment:
'postprocessing the images' -- which postprocessing methods? I think you mean postprocessing the predictions?

Reply:
'private leaderboard performance is a blackbox until the competition deadline' The term blackbox in engineering refers to something where you know the inputs and outputs, but don't know how the computation takes place. In that sense, I don't think the private leaderboard performance is a blackbox -- it's just unknown.

Reply:
'When the model approaches an accuracy that is hampered by the noise of the dataset itself, progression towards an incrementally better score can quickly become time, resource and computationally expensive.' I'm a little confused by this sentence. I think you're trying to say that it takes great effort to make a small improvement in the score once you have a decent model.